iCount/analysis/rnamaps.py

```diff
@@ -3,690 +3,233 @@
-
+Perform RNA-maps analysis.
 """
+import csv
 import logging
+import os
 
-import pybedtools
-import matplotlib
-matplotlib.use('Agg')
-import matplotlib.pyplot as plt  # pylint: disable=wrong-import-position
+from pybedtools import BedTool, create_interval_from_list
 
 import iCount  # pylint: disable=wrong-import-position
-from iCount.files import _f2s  # pylint: disable=wrong-import-position
 
 LOGGER = logging.getLogger(__name__)
 
-EXON_TYPES = ['CDS', 'ncRNA', 'UTR3', 'UTR5']
-RNA_WINDOW_SIZE = 2000  # TODO: rethink how this constant would affect lengths in RNAmap generation
-
-
-def _add_entry(start_type, stop_type, distance, score, strand, data, metrics, explict=False):
-    """Add RNA-map entry in ``data``."""
-    if strand == '+':
-        rna_map_type = start_type + '-' + stop_type
-    else:
-        rna_map_type = stop_type + '-' + start_type
-        distance = -distance
-
-    data.setdefault(rna_map_type, {}).setdefault(distance, [0, 0])[0] += score
-    if explict:
-        data.setdefault(rna_map_type, {}).setdefault(distance, [0, 0])[1] += score
-
-    # Increment also mRNA / pre-mRNa counters:
-    if 'intron' in rna_map_type or 'intergenic' in rna_map_type:
-        metrics.origin_premrna += score
-    elif start_type in EXON_TYPES and stop_type in EXON_TYPES:
-        metrics.origin_mrna += score
-    else:
-        metrics.origin_ambiguous += score
-
-
-def _process_read_group(xlink, chrom, strand, read, data, segmentation, metrics,
-                        implicit_handling='closest'):
-    """
-    Process each read group.
-
-    Each group is represented by read. It has a fixed xlink, start, second-start
-    and end position. This function finds all landmarks that this read is
-    overlapping. It also determines for each of them:
+MAP_TYPES = {
+    'exon-intron': {
+        'upstream-type': ['CDS'],
+        'upstream-size-limit': 50,
+        'upstream-plot-width': 50,
+        'downstream-type': ['intron'],
+        'downstream-size-limit': 150,
+        'downstream-plot-width': 100,
+    },
+    'intron-exon': {
+        'upstream-type': ['intron'],
+        'upstream-size-limit': 150,
+        'upstream-plot-width': 100,
+        'downstream-type': ['CDS'],
+        'downstream-size-limit': 50,
+        'downstream-plot-width': 50,
+    },
+    'gene-start': {
+        'upstream-type': ['intergenic'],
+        'upstream-plot-width': 200,
+        'downstream-type': ['UTR5', 'CDS'],
+        'downstream-plot-width': 50,
+    },
+    'gene-end': {
+        'upstream-type': ['CDS', 'UTR3'],
+        'upstream-plot-width': 200,
+        'downstream-type': ['intergenic'],
+        'downstream-plot-width': 200,
+    },
+    'translation-start': {
+        'upstream-type': ['UTR5'],
+        'upstream-plot-width': 50,
+        'downstream-type': ['intron', 'CDS'],
+        'downstream-plot-width': 50,
+    },
+    'translation-end': {
+        'upstream-type': ['intron', 'CDS'],
```

Comment:
@JureZmrzlikar correct to: 'downstream-type': ['UTR3']

Reply:
It already is `'downstream-type': ['UTR3']`. Did you meant something else?